File: 03-Code/Block/06_RetainCycle_MRC/Block01/Person.m
//
//  Person.m
//  Block01
//
//  Created by min Lee on 2018/6/15.
//  Copyright © 2018年 min Lee. All rights reserved.
//

#import "Person.h"

@implementation Person

- (void)dealloc {
    [super dealloc];
    NSLog(@"Person-dealloc");
}

- (void)test {
    //使用__unsafe_unretained解决
//    __unsafe_unretained id weakSelf = self;
//    self.block = ^{
//        NSLog(@"%p", weakSelf);
//    };
    
    
    //使用__block解决
    __block id weakSelf = self;
    self.block = ^{
        NSLog(@"%p", weakSelf);
    };
}

@end

File: 03-Code/SwiftAppUsingOC/SwiftAppUsingOC/Person.swift
//
//  SwiftTest.swift
//  SwiftAppUsingOC
//
//  Created by min Lee  on 2024/3/20.
//  Copyright © 2024 min Lee. All rights reserved.
//

import Foundation

// 必须继承 NSObject
class Person: NSObject {
    // 公开给OC的要使用 @objc 修饰
    @objc var name: String
    @objc var age : Int

    @objc init(name: String, age: Int) {
        self.name = name
        self.age = age
        print("Person init--name:\(name)；age=\(age)")
    }
    
    @objc func getInfo() {
        print("self.name：\(self.name),self.age = \(self.age)")
    }
    
    @objc class func classMethod() {
        print("Person classMethod")
    }
    
    @objc static func staticMethod() {
        print("Person staticMethod")
    }
}

